### PR TITLE
GH-41584: [Java] ListView Implementation for C Data Interface

### PR DESCRIPTION
**Pull request by vibhatha**
### Rationale for this change

C Data Interface is missing `ListView` and `LargeListView` after recently merging core functionalities.

Also closes;

- [x] https://github.com/apache/arrow/issues/41585

### What changes are included in this PR?

This PR includes C Data interface related component additions to `ListView` and `LargeListView` along with the corresponding test cases. 

### Are these changes tested?

Yes

### Are there any user-facing changes?

No
* GitHub Issue: #41584

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/complex/BaseLargeRepeatedValueViewVector.java [hunk 1]
```diff
@@ -311,17 +311,14 @@ public void setValueCount(int valueCount) {
...
       int currentOffset = offsetBuffer.getInt((long) i * OFFSET_WIDTH);
       int currentSize = sizeBuffer.getInt((long) i * SIZE_WIDTH);
       int currentSum = currentOffset + currentSize;
-
       maxOffsetSizeSum = Math.max(maxOffsetSizeSum, currentSum);
-      minOffset = Math.min(minOffset, currentOffset);
     }
 
-    return maxOffsetSizeSum - minOffset;
+    return maxOffsetSizeSum;
```
Comment (vibhatha) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/BaseLargeRepeatedValueViewVector.java:321]: @lidavidm what I have done in the first implementation is incorrect, I think the test case I had didn't tackle this case, but the C data interface tests caught it. Sorry about that.

Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseLargeRepeatedValueViewVector.java:321]: This need a little bit of work... I am improving it.

Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseLargeRepeatedValueViewVector.java:327]: I updated.

File: java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java [hunk 2]
```diff
@@ -304,38 +304,44 @@ public void setValueCount(int valueCount) {
...
       reallocateBuffers();
     }
-    final int childValueCount = valueCount == 0 ? 0 : getLengthOfChildVector();
+    final int childValueCount = valueCount == 0 ? 0 : getMaxViewEndChildVector();
     vector.setValueCount(childValueCount);
   }
 
-  protected int getLengthOfChildVector() {
+  /**
+   * Get the maximum view length of child vector This method deduces the length by considering the
```
[the remark this answers is not in the document]
Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java:312]: Right, the function name is correct but the docs are not precise. I will update.

[the remark this answers is not in the document]
Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java]: I updated the docs, does it look okay?

File: java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java [hunk 3]
```diff
@@ -304,38 +304,44 @@ public void setValueCount(int valueCount) {
...
+   * condition i.e., 0 <= offsets[i] + size[i] <= length of the child array.
```
[the remark this answers is not in the document]
Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java]: Updated this as well.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**